Ancestors.java

```diff
@@ -0,0 +1,42 @@
+import java.util.ArrayList;
+
+/**
+ * Created by ioana-chirca on 28-May-17.
+ */
+public class Ancestors<T> {
+    public ArrayList<T> printAncestors(BinaryTree<T> tree, T key) {
+        ArrayList<T> ancs = new ArrayList<>();
+        ancestors(tree.getRoot(), key, ancs);
+
+        return ancs;
+    }
+
+    private boolean ancestors(BinaryTree.Node current, T key, ArrayList<T> ancs) {
```

Comment:
It would be more readable if you named ancs parameter ancestors or ancestorsList. And change the method's name to reflect what it does: something like buildAncestorsList or so.